hal/sdio_api.h

```diff
@@ -0,0 +1,163 @@
+ * @retval SD status
+ */
+int sdio_init(void);
+
+/**
+ * @brief  DeInitializes the SD card device.
+ * @retval SD status
+ */
+int sdio_deinit(void);
+
+/**
+ * @brief  Reads block(s) from a specified address in an SD card, in polling mode.
+ * @param  data: Pointer to the buffer that will contain the data to transmit
+ * @param  address: Address from where data is to be read
+ * @param  block_count: Number of SD blocks to read
+ * @retval SD status
+ */
+int sdio_read_blocks(uint8_t *data, uint32_t address, uint32_t block_count);
+
+/**
+ * @brief  Writes block(s) to a specified address in an SD card, in polling mode.
+ * @param  data: Pointer to the buffer that will contain the data to transmit
+ * @param  address: Address from where data is to be written
+ * @param  block_count: Number of SD blocks to write
+ * @retval SD status
+ */
+int sdio_write_blocks(uint8_t *data, uint32_t address, uint32_t block_count);
+
+#if DEVICE_SDIO_ASYNC
+
+/**
+ * @brief  Reads block(s) from a specified address in an SD card, in DMA mode.
+ * @param  data: Pointer to the buffer that will contain the data to transmit
+ * @param  address: Address from where data is to be read
+ * @param  block_count: Number of SD blocks to read
+ * @retval SD status
+ */
+int sdio_read_blocks_async(uint8_t *data, uint32_t address, uint32_t block_count);
+
+/**
+ * @brief  Writes block(s) to a specified address in an SD card, in DMA mode.
+ * @param  data: Pointer to the buffer that will contain the data to transmit
+ * @param  address: Address from where data is to be written
+ * @param  block_count: Number of SD blocks to write
+ * @retval SD status
+ */
+int sdio_write_blocks_async(uint8_t *data, uint32_t address, uint32_t block_count);
+
+/**
+ * @brief  Check if a DMA operation is pending
+ * @retval DMA operation is pending
+ *          This value can be one of the following values:
+ *            @arg  SD_TRANSFER_OK: No data transfer is acting
+ *            @arg  SD_TRANSFER_BUSY: Data transfer is acting
+ */
+int sdio_read_pending(void);
+
+/**
+ * @brief  Check if a DMA operation is pending
+ * @retval DMA operation is pending
+ *          This value can be one of the following values:
+ *            @arg  SD_TRANSFER_OK: No data transfer is acting
+ *            @arg  SD_TRANSFER_BUSY: Data transfer is acting
+ */
+int sdio_write_pending(void);
+
+#endif // DEVICE_SDIO_ASYNC
+
+/**
+ * @brief  Erases the specified memory area of the given SD card.
+ * @param  start_address: Start byte address
+ * @param  end_address: End byte address
+ * @retval SD status
+ */
+int sdio_erase(uint32_t start_address, uint32_t end_address);
+
+/**
+ * @brief  Gets the current SD card data status.
+ * @param  None
+ * @retval Data transfer state.
+ *          This value can be one of the following values:
+ *            @arg  SD_TRANSFER_OK: No data transfer is acting
+ *            @arg  SD_TRANSFER_BUSY: Data transfer is acting
+ */
+int sdio_get_card_state(void);
```

Comment:
In my opinion the documentation tells that this would be the same as checking sdio_read_pending and sdio_write_pending in a row. Implementation seems to be something different. Would you please update this comment section.